neo4j/internal/packstream/packer.go

```diff
@@ -127,7 +127,7 @@ func (p *Packer) writeListHeader(l int, shortOffset, longOffset byte) error {
 			hdr = hdr[:1+2]
 			hdr[0] = longOffset + 1
 			binary.BigEndian.PutUint16(hdr[1:], uint16(l))
-		case l < math.MaxUint32:
+		case uint32(l) < uint32(math.MaxUint32):
```

Comment:
uint32(l) will truncate l to fit into uint32 and the comparison will always succeed
l needs to be int64 instead of int to compile and work properly on 32/64 bits systems.

Reply:
Thanks for the feedback! Didn't know that `int64` [would work](https://groups.google.com/forum/#!topic/golang-nuts/mtnn-01Dh_I) on a 32-bit system